Nil pointer dereference error on `osm install`  when the `ca.crt` or `private.key` is configured incorrectly

<!--

This issue tracker is a best-effort forum for users and customers to report bugs.

Be sure to not include any sensitive information. Sensitive information should __NOT__ be included in this issue.

-->

**Bug description**:

If the ca cert or private key contained in the CA bundle secret is configured incorrectly and Tresor is the cert provider, a segmentation fault will occur when trying to issue a cert due to a nil pointer dereference. 

This scenario occurred when a customer attempted to bring their own root cert and the private key was not formatted correctly. #4458 

After an error occurs when decoding the ca cert or private key, the certificate issuing function should return the error to avoid the noisy segfault.

osm injector error log:
```
{"level":"info","component":"osm-injector/main","time":"2022-01-18T18:31:58Z","file":"osm-injector.go:94","message":"Starting osm-injector dev; 7df9bad; "}
W0118 18:31:58.551460 1 client_config.go:614] Neither --kubeconfig nor --master was specified. Using the inClusterConfig. This might not work.
{"level":"info","component":"osm-injector/main","time":"2022-01-18T18:31:58Z","file":"osm-injector.go:142","message":"Initial MeshConfig osm-mesh-config: {\n "sidecar": {\n "logLevel": "error",\n "envoyImage": "envoyproxy/envoy-alpine:v1.18.3",\n "initContainerImage": "openservicemesh/init:v0.9.2",\n "configResyncInterval": "0s",\n "resources": {}\n },\n "traffic": {\n "inboundExternalAuthorization": {\n "statPrefix": "inboundExtAuthz",\n "timeout": "1s"\n }\n },\n "observability": {\n "osmLogLevel": "info",\n "tracing": {\n "port": 9411,\n
"address": "jaeger.osm-system.svc.cluster.local",\n "endpoint": "/api/v2/spans"\n }\n },\n "certificate": {\n "serviceCertValidityDuration": "24h"\n },\n "featureFlags": {\n "enableWASMStats": true,\n "enableEgressPolicy": true\n }\n}"}
{"level":"info","component":"kube-controller","time":"2022-01-18T18:31:58Z","file":"client.go:139","message":"Namespace controller client started"}
{"level":"info","component":"kube-controller","time":"2022-01-18T18:31:58Z","file":"client.go:154","message":"Waiting for Namespaces informer cache sync..."}
{"level":"info","component":"kube-controller","time":"2022-01-18T18:31:58Z","file":"client.go:164","message":"Caches for [Namespaces] synced successfully"}
{"level":"info","component":"cert-provider-util","time":"2022-01-18T18:31:58Z","file":"config.go:184","message":"CA already exists in kubernetes, loading."}
{"level":"error","component":"tresor","error":"no certificate in PEM","time":"2022-01-18T18:31:59Z","file":"certificate_manager.go:60","message":"Error decoding Root Certificate's Private Key PEM "}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6ff905]
goroutine 1 [running]:
crypto/rsa.(*PrivateKey).Public(0x0, 0x1b8cce0, 0x0)
/opt/hostedtoolcache/go/1.15.15/x64/src/crypto/rsa/rsa.go:112 +0x5
crypto/x509.CreateCertificate(0x1f2e600, 0xc0001231d0, 0xc0007ad3e0, 0xc00037a680, 0x1a79aa0, 0xc0007e34a0, 0x1b8cce0, 0x0, 0xc000a46cc0, 0x4, ...)
/opt/hostedtoolcache/go/1.15.15/x64/src/crypto/x509/x509.go:2103 +0xb7
github.com/openservicemesh/osm/pkg/certificate/providers/tresor.(*CertManager).issue(0xc0007e33e0, 0xc000a46f20, 0x1b, 0x460623fc85e0000, 0x0, 0xc000723970, 0x4f1bf1, 0x2c00ca0)
/home/runner/work/osm/osm/pkg/certificate/providers/tresor/certificate_manager.go:63 +0x3eb
github.com/openservicemesh/osm/pkg/certificate/providers/tresor.(*CertManager).IssueCertificate(0xc0007e33e0, 0xc000a46f20, 0x1b, 0x460623fc85e0000, 0x2, 0xc000a46f20, 0x1b, 0x1f89a60)
/home/runner/work/osm/osm/pkg/certificate/providers/tresor/certificate_manager.go:120 +0xca
github.com/openservicemesh/osm/pkg/injector.NewMutatingWebhook(0x2382, 0x1fa6fe0, 0xc000152f20, 0x1f89a00, 0xc0007e33e0, 0x1f968a0, 0xc000545530, 0x7ffc5aab19ae, 0x3, 0x7ffc5aab1997, ...)
/home/runner/work/osm/osm/pkg/injector/webhook.go:70 +0x116
main.main()
/home/runner/work/osm/osm/cmd/osm-injector/osm-injector.go:161 +0x9e5
```

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [x]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**:

The logs should not contain the segfault due to a nil pointer expectation.

osm bootstrap error logs with no segfault:
```shell
{"level":"info","component":"cert-provider-util","time":"2022-01-20T01:52:44Z","file":"config.go:190","message":"CA already exists in kubernetes, loading."}
{"level":"error","component":"tresor","error":"no certificate in PEM","error_code":"E4012","time":"2022-01-20T01:52:45Z","file":"certificate_manager.go:76","message":"Error decoding Root Certificate's Private Key PEM"}
{"level":"fatal","component":"kube-events","error":"Error issuing certificate for the crd-converter: no certificate in PEM","reason":"FatalInitializationError","time":"2022-01-20T01:52:45Z","file":"events.go:140","message":"Error creating crd conversion webhook"}
```

**Steps to reproduce the bug (as precisely as possible)**:
As described by the customer in issue #4458:

I am using the below openssl command to generate the root cert:

**Generate cert**
openssl genrsa -out cakey.pem 2048 # generate root key
openssl req -x509 -new -nodes -key cakey.pem -sha256 -days 2000 -out cacert.pem # generate root cert

**create secret in osm namespace**
kubectl create secret generic osm-ca-bundle --from-file=ca.crt=cacert.pem --from-file=private.key=cakey.pem -n osm-system (add expiration as well)

**deploy osm using helm**
helm install osm osm --repo https://openservicemesh.github.io/osm --version 0.11 --namespace osm-system --values values.yaml

**How was OSM installed?**:
Helm

**Anything else we need to know?**:

**Bug report archive**:
<!--
Generate and upload bug report archive for the given app namespaces, deployments, and pods.
Use `osm support bug-report --app-namespaces <ns1>,<ns2> --app-deployments <ns1>/<dep1>,<ns2>/<dep2> --app-pods <ns1>/<pod1>`
-->

**Environment**:
- OSM version (use `osm version`):0.11
- Kubernetes version (use `kubectl version`):1.21
- Size of cluster (number of worker nodes in the cluster):
- Others:

--- a/pkg/certificate/providers/tresor/certificate_manager.go
+++ b/pkg/certificate/providers/tresor/certificate_manager.go
@@ -81,6 +81,7 @@ func (cm *CertManager) IssueCertificate(cn certificate.CommonName, validityPerio
 		// TODO(#3962): metric might not be scraped before process restart resulting from this error
 		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrDecodingPEMCert)).
 			Msg("Error decoding Root Certificate's PEM")
+		return nil, errors.Wrap(err, errCreateCert.Error())
 	}
 
 	rsaKeyRoot, err := certificate.DecodePEMPrivateKey(cm.ca.GetPrivateKey())
@@ -88,6 +89,7 @@ func (cm *CertManager) IssueCertificate(cn certificate.CommonName, validityPerio
 		// TODO(#3962): metric might not be scraped before process restart resulting from this error
 		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrDecodingPEMPrivateKey)).
 			Msg("Error decoding Root Certificate's Private Key PEM ")
+		return nil, errors.Wrap(err, errCreateCert.Error())
 	}
 
 	derBytes, err := x509.CreateCertificate(rand.Reader, &template, x509Root, &certPrivKey.PublicKey, rsaKeyRoot)

